packages/core/src/common/env-variables/env-variables-protocol.ts

```diff
@@ -21,6 +21,10 @@ export interface EnvVariablesServer {
     getExecPath(): Promise<string>
     getVariables(): Promise<EnvVariable[]>
     getValue(key: string): Promise<EnvVariable | undefined>
+    getUserHomeFolderPath(): Promise<string>
+    getDataFolderName(): Promise<string>
+    getUserDataFolderPath(): Promise<string>
+    getAppDataPath(): Promise<string>
```

Comment:
If this API is available from the browser too (as I [see](https://github.com/eclipse-theia/theia/pull/6650/files#diff-110a68fbfaa2b58be0ae99da01834e4aR39) yes), I'd propose having `getUserHomeFolderUri`, `getUserDataFolderUri`, and `getAppDataPath`. It is an anti-pattern to pass an FS paths as a string between the frontend and the backend. The Frontend cannot do much with it, can it?

Reply:
Also, `getDataFolderName()` can be derived from ~`getUserDataFolderPath`~ `getUserDataFolderUri`. This method is unnecessary.

Reply:
@kittaakos I really don't know what is better approach here. If we change these methods to return URI, then, for consistence, we have to do the same on backend. But on backend that values is used mostly as strings and using URI there seems like overhead to me (and looks awkward). But maybe I am wrong... @akosyakov @svenefftinge what's your opinion?

Reply:
> we have to do the same on backend

Correct. And we have clear API for it: `FileUri.fsPath`.

> (and looks awkward)

All resource info is exchanged as a URI string in Theia, if no, that API is flawed. If for whatever reason you need an FS path on the FE, use `fileSystem.fsPath(uri)`.

Reply:
overlooked it, going with URIs will be more consistent indeed, fsPath can be always derived from them on backend